test/models/drive_test.dart

```diff
@@ -0,0 +1,1003 @@
+      final List<Drive> drives = Drive.fromJsonList(jsonList);
+      expect(drives.length, 3);
+      expect(drives.first.id, json['id']);
+      expect(drives[1].start, json['start']);
+      expect(drives.last.seats, 2);
+    });
+    test('can handle an empty list', () {
+      final List<Drive> drives = Drive.fromJsonList([]);
+      expect(drives, []);
+    });
+  });
+  group('Drive.toJson', () {
+    test('returns a json representation of the drive', () async {
+      final Drive drive = DriveFactory().generateFake();
+      final Map<String, dynamic> json = drive.toJson();
+      expect(json['start'], drive.start);
+      expect(json['start_lat'], drive.startPosition.lat);
+      expect(json['start_lng'], drive.startPosition.lng);
+      expect(json['start_time'], drive.startTime.toString());
+      expect(json['end'], drive.end);
+      expect(json['end_lat'], drive.endPosition.lat);
+      expect(json['end_lng'], drive.endPosition.lng);
+      expect(json['end_time'], drive.endTime.toString());
+      expect(json['cancelled'], drive.cancelled);
+      expect(json['seats'], drive.seats);
+      expect(json['driver_id'], drive.driverId);
+      expect(json['hide_in_list_view'], drive.hideInListView);
+      expect(json.keys.length, 12);
+    });
+  });
+  group('Drive.approvedrides', () {
+    test('return null if rides is null', () async {
+      final Drive drive = DriveFactory().generateFake(
+        createDependencies: false,
+      );
+      expect(drive.approvedRides, null);
+    });
+    test('return an empty List if rides has no approved rides', () async {
+      final Drive drive = DriveFactory().generateFake(
+        rides: [
+          RideFactory().generateFake(status: RideStatus.pending),
+        ],
+      );
+      expect(drive.approvedRides, []);
+    });
+    test('return a ride in a List, if rides only has 1 ride and it is approved', () async {
+      final Ride ride = RideFactory().generateFake(status: RideStatus.approved);
+      final Drive drive = DriveFactory().generateFake(rides: [ride]);
+      expect(drive.approvedRides, [ride]);
+    });
+    test('return correct ride with rides having 1 approved one in it', () async {
+      final Ride ride = RideFactory().generateFake(status: RideStatus.approved);
+      final Drive drive = DriveFactory().generateFake(
+        rides: [
+          RideFactory().generateFake(status: RideStatus.pending),
+          RideFactory().generateFake(status: RideStatus.cancelledByDriver),
+          RideFactory().generateFake(status: RideStatus.pending),
+          ride,
+          RideFactory().generateFake(status: RideStatus.rejected),
+          RideFactory().generateFake(status: RideStatus.cancelledByRider),
+          RideFactory().generateFake(status: RideStatus.withdrawnByRider),
+        ],
+      );
+      expect(drive.approvedRides, [ride]);
+    });
+    test('return correct rides with rides have more than 1 approved ones in it', () async {
```

Comment:
Please keep the test cases minimal. They should document the code and be readable. Every test case should be specific. Here, for example, there is no need for one Ride of every other status to be in the list, since it that has already been tested in the test above.